### PR TITLE
Add ugrid support

### PR DESCRIPTION
**Pull request by ChristosT**
This PR adds support for the AFLR UGRID file format. The UGRID file format is described in detail in http://www.simcenter.msstate.edu/software/downloads/doc/ug_io/index.html .  
Among others it is used in AFLR3 grid generator and in NASA's FUND3D flow solver. 

This PR adds ability to read and write meshes in C-binary , FORTRAN-formatted and ASCII format.
Tests added for all formats as well as for testing the node ordering of pyramids which differs from meshio native ordering.

### CODE REVIEW

File: meshio/ugrid/_ugrid.py [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+        type_suffix = filename_parts[-2]
+        if type_suffix in file_types.keys():
+            file_type = file_types[type_suffix]
+
+    return file_type
+
+
+def read(filename):
+
+    global file_type
```
Comment (nschloe) [inline meshio/ugrid/_ugrid.py]: Say no to globals.

File: meshio/ugrid/_ugrid.py [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+
+    with open_file(filename) as f:
+        mesh = read_buffer(f)
+    return mesh
+
+
+def _read_section(f, count, dtype):
+    if file_type["type"] == "ascii":
+        return numpy.fromfile(f, count=count, dtype=dtype, sep=" ")
+    else:
```
Comment (nschloe) [inline meshio/ugrid/_ugrid.py]: This `else` isn't necessary.

File: meshio/ugrid/_ugrid.py [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+    if file_type["type"] == "ascii":
+        ncols = array.shape[1]
+        fmt = " ".join(["%r"] * ncols)
+        numpy.savetxt(f, array, fmt=fmt)
+    else:
+        array.astype(dtype).tofile(f)
+
+
+def write(filename, mesh):
+    global file_type
```
Comment (nschloe) [inline meshio/ugrid/_ugrid.py]: Again. :smile_cat:

File: meshio/ugrid/_ugrid.py (hunk 1 above)
Comment (ChristosT) [reply meshio/ugrid/_ugrid.py]: Sure !

File: meshio/ugrid/_ugrid.py (hunk 2 above)
Comment (ChristosT) [reply meshio/ugrid/_ugrid.py]: I am not sure I understand this.  Could you elaborate on it ? I just followed the vtk writer/reader
from here https://github.com/nschloe/meshio/blob/049c403eefbf339a2b8e8f7474c4f9210ea67f00/meshio/vtk/_vtk.py#L349

Comment (nschloe) [reply meshio/ugrid/_ugrid.py]: Just remove else and unindent the block.

File: meshio/ugrid/_ugrid.py (hunk 3 above)
Comment (ChristosT) [reply meshio/ugrid/_ugrid.py]: Done :smiley:

File: meshio/ugrid/_ugrid.py (hunk 2 above)
Comment (ChristosT) [reply meshio/ugrid/_ugrid.py]: Yep done in the latest commit

**Comment by nschloe** [comment]
LGTM, thanks!

**Pull request merged**